votor/src/timer_manager/timers.rs

```diff
@@ -98,20 +99,24 @@ pub(super) struct Timers {
     heap: BinaryHeap<Reverse<(Instant, Slot)>>,
     /// Channel to send events on.
     event_sender: Sender<VotorEvent>,
+    /// Stats for the timer manager.
+    stats: Arc<Mutex<TimerManagerStats>>,
```

Comment:
Would it make sense to split this instead:
- set_timeouts fills/returns metrics about timer insertion for reporting in EventHandler
- progress fills/returns metrics about heap size / timers popped for reporting in SkipTimerService
that way we could avoid having 2 threads updating the same stats and using a mutex?

Reply:
Oh I meant heap size is only reported from progress, is it necessary to report it from set_timeouts as well?

Reply:
set_timeout pushes to heap, so it's actually the more important place. We could only report heap from there, but then we don't get update when heap is popped in progress.

Reply:
Under normal cases it's fine, if we don't have set_timeout() in 10 seconds then this number isn't accurate any more, but I guess we don't care that much about this case.